### DISCUSSION TITLE
Player stats not saving to DB

### DISCUSSION THREAD

**Discussion by b4guette1** [jacobwojoski/dice-stats#88] [Q&A] [answered]
I have one player who's stats won't save to DB ever. Like the rest of them all save as they roll and this one player has their dice rolls never saving to the DB, ever. Even when they save voluntarily does it not work. Or at least as the GM I can't see them.

**Comment by jacobwojoski** [comment]
Hard to say without looking at it. Have you tried clearing the data base? Usually clearing all the data from the global stats fixes most DB issues.

**Comment by b4guette1** [comment]
i'll try that and update on how it goes

**Reply by jacobwojoski** [reply]
If that doesn’t work have the player press f12 and press the console button. 

check the console for any errors when rolling a die. If they exits a screenshot would help.

**Reply by NachoArcade** [reply]
Hey, I have a similar issue where the roll stats save to the database while the player is in the game, but once they leave it does not load when they rejoin. The main error I can see occur during this is the following:
[Error.log](https://github.com/jacobwojoski/dice-stats/files/13798937/Error.log)

I am unsure if this is the exact issue causing it or not.

This is the ErrorDieData log from when one of the players load into the game:
[ErrorDieData.log](https://github.com/jacobwojoski/dice-stats/files/13798958/ErrorDieData.log)

**Reply by jacobwojoski** [reply]
@NachoArcade what game system are you using?

**Reply by NachoArcade** [reply]
I may have responded incorrectly so that you missed it. @jacobwojoski so in the event that you did, we are using A5e.
![image](https://github.com/jacobwojoski/dice-stats/assets/27779832/d6730aae-1f43-4669-9b79-063e6c6f2d6b)

Its been awhile since I checked in due to being preoccupied with real life stuff and campaign being on hiatus due to it which is why I am getting back to this now, although since we are resuming I am curious if you had any idea what might cause this issue,

**Comment by NachoArcade** [comment]
> @NachoArcade what game system are you using?

We are using Level Up: Advanced 5th Edition (A5e)

**Reply by jacobwojoski** [reply, marked as answer]
Sorry totally had this on my back burner and forgot about it.

**TRY THIS FIRST**
Looks like its throwing the error 'Error thrown in hooked function '' for hook 'ready'. dbDieDataObj.MEANS' in the log. The usual fix for this is to clear the database and reload the system. There was an update a while back that slightly changed how I did the database stuff and a reset is needed.

**Look into this if part 1 doesn't work**
There seems to also be an error finding an object called 'clearAllTiles' from the buttons on the left (Scene control buttons). That's def some issue from another mod. And could be causing an issue trying to add his icon to the list. Somethings adding the clearAllTiles item incorrectly. As I don't think that button it default to foundry so you would need to find the module that adds that button and get back to me.

**DEF TRY RESETTING THE DATABASE FROM THE GM SIDE, GM->DICE_STATS->GLOBAL_DICE_STATS->CLEAR ALL (DB AND PLAYER)**
